### PR TITLE
#29 create trecipe page 

### PR DESCRIPTION
**Pull request by shizuko-akamoto**
Create the Trecipe Page using all the components we have developed so far:

### **1. Header component**
Header component with logo, nav bar and the search bar

### **2. Footer component**
Footer component is fixed at the bottom, and appears when the user scrolls all the way to the bottom. Still have to create more contents for the footer.

### **3. MyTrecipes page**
Most of the components are currently for display only. The only functionality that's actually placed in is the "create new trecipe" workflow. The name, description, and privacy setting input from AddPopup will be reflected on the new TC created. 

**Responsiveness of the page**
I used css grids to handle fitting the TC when scaling window sizes. There is a minimum width set for TC (350px), and once the width becomes small enough, the grid column number will be reduced by 1. 

**Demo capture**
Trecipe-Google-Chrome-2020-06-07-17-08-59

After addressing styling feedback:
image

### CODE REVIEW

**Comment by tianyan-zhu** [comment]
I agree with @knox153, that round corner cards will look better on the my Trecipe page ;)

File: src/components/Footer.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    private static githubLink: string = 'https://github.com/shizuko-akamoto/Trecipe';
+
+    render() {
+        return (<footer className='footer'>
+            <div className="footer-contents">
+                <div className='footer-logo'>
+                    <h2>Trecipe</h2>
+                    <p>CPSC432I(2020S) Team2</p>
+                </div>
+                <div className='sns'>
```
[the remark this answers is not in the document]
Comment (shizuko-akamoto) [reply src/components/Footer.tsx:15]: Social Networking Service LOL.
jk it's garbage from a tutorial I was following :P 
Renamed to `links`

File: src/components/TrecipeCard.tsx [hunk 2]
```diff
@@ -24,39 +25,44 @@ export interface TCProps {
...
-                <div className="Text">
-                    <div className='MetaData'>
-                        <div className='date'>{this.props.date}</div>
-                        <div className='author'>by: {this.props.author}</div>
+            <div className='trecipeCard'>
+                <div className="tcHeaderContainer" style={{backgroundImage: this.props.imageSrc}}>
+                    <div className="tcHeader">
+                        <label className="tcTitle">
+                            {this.props.name}
+                            <FontAwesomeIcon icon={(this.props.isPrivate) ? "lock" : "unlock"} className='tcPrivacy'/>
```
[the remark this answers is not in the document]
Comment (shizuko-akamoto) [reply src/components/TrecipeCard.tsx:51]: personally, i like having the public icon. I found an alternative one to replace the current one.
image
Hopefully this is a little clearer 😄

File: src/stylesheets/variables.scss [hunk 3]
```diff
@@ -20,8 +20,11 @@ $default-font-family: 'Lato', sans-serif;
...
+$logo-font-size: 4rem;
 $button-font-size: 0.875rem;
 $input-font-size: 0.8rem;
 $default-font-size: 1rem;
-$title-font-size: 2rem;
\ No newline at end of file
+$title-font-size: 2rem;
+
+// --- Element Widths ---//
+$tc-max-width: 350px;
```
[the remark this answers is not in the document]
Comment (shizuko-akamoto) [reply src/stylesheets/variables.scss:30]: Yup i agree 😅

**Comment by shizuko-akamoto** [comment]
> Just noticed there is [flash of unstyled content](https://en.wikipedia.org/wiki/Flash_of_unstyled_content) on the Trecipe logo. If you refresh the page, the Trecipe logo on the top left has a different font-family for a brief moment before the correct one shows up. This is a tinyyyy issue, maybe we can log a issue and put it in the backlog for now.
> 
> Not sure if these changes below would make it looks better 👇
> 
> feeback
> 
> Ohhh and I will wait until this branch is merged before merging mine.

I can't reproduce the issue on my end... but we can investigate a little on component lifecycle methods to see if there's one that let us wait on the rendering.... but yea, to be addressed in future 😆

File: src/components/TrecipeCard.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (knox153) [reply src/components/TrecipeCard.tsx]: Haha that is much better, thank you :+1:

**Comment by shizuko-akamoto** [comment]
> Good to go 🎉
> Linking the unstyled content issue: #53

woo. thanks for creating the issue

**Pull request merged**